server/aesdsocket.c

```diff
@@ -0,0 +1,258 @@
+    socklen_t addr_size;
+
+    memset(&hints, 0 , sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+    hints.ai_protocol = 0;
+
+    get_fd = getaddrinfo(NULL, "9000", &hints, &rp);
+    if(get_fd != 0)
+    {
+        perror("ERROR");
+        exit(1);
+    }
+    
+    int d_flag = 0;
+  if(argc == 2)
+    {
+        if((strcmp(argv[1],"-d")) == 0)
+        {
+            d_flag = 1;
+        }
+
+        else
+        {
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    server_socket = socket(rp->ai_family, rp->ai_socktype, rp->ai_protocol);
+    if(server_socket == -1)
+    {
+        perror("ERROR");
+        exit(1);
+    }
+    if(setsockopt(server_socket,SOL_SOCKET, SO_REUSEADDR, &(int){1}, sizeof(int))<0)
+        perror("setsocket(SO_REUSEADDR) failed");
+
+    //bind the socket to our specified IP and port
+    if(bind(server_socket, rp->ai_addr, rp->ai_addrlen) != 0)
+    {
+        perror("ERROR");
+        exit(1);
+    }
+    syslog(LOG_DEBUG, "Binded successfully");
+    freeaddrinfo(rp);
+
+    if(d_flag == 1)
+    {
+        daemonise_process();
+    }
+    
+    //File to store the recieved data
+    int store_fd;
+    store_fd = open("/var/tmp/aesdsocketdata", O_RDWR|O_CREAT|O_TRUNC, 0666);
+    if(store_fd < 0)
+    {
+        perror("open");
+    }
+
+    //Server socket is established, next Client-Server connection should be made
+    while(1){
+    listen(server_socket, 5);
+    
+    
+    addr_size = sizeof(cl_addr);
+    client_socket = accept(server_socket, (struct sockaddr *)&cl_addr, &addr_size);
+    struct sockaddr_in* pv4Addr = (struct sockaddr_in*)&cl_addr;
+    struct in_addr ipAddr = pv4Addr->sin_addr;
+    char str[INET_ADDRSTRLEN];
+    inet_ntop( AF_INET, &ipAddr,str, INET_ADDRSTRLEN);
+    syslog(LOG_DEBUG, "Accepted connection to %s\n", str);
+
+    //Client-Server connection is made.
+    char *recv_buf = (char *)malloc(1024);
+    int realloc_cnt = 1;
+    int recv_bytes = 0;
+    int recv_buf_size = 0;
+    int written_bytes = 0;
+    if(recv_buf == NULL)
+    {
+        close(client_socket);
+        exit(-1);
+    }
+
+    
+    //read bytes
+    while(1)
+    {
+        recv_bytes = recv(client_socket, &recv_buf[recv_buf_size], 1024, 0);
+        recv_buf_size += recv_bytes;
+        if(!recv_bytes)
+        {
+            break;
+        }
+        if(recv_bytes < 1024)
+        {
+            if(recv_buf[recv_buf_size-1] == '\n')
+            {
+                written_bytes = write(store_fd, recv_buf, recv_buf_size);
+                if(written_bytes < 0)
+                {
+                    perror("write");
+                }
+                free(recv_buf);
+                recv_buf = NULL;
+                recv_buf_size = 0;
+                break;
+            }
+        }
+        else
+        {
+            realloc_cnt++;
+            recv_buf = realloc(recv_buf, realloc_cnt * 1024);
```

Comment:
@Jahnavi-pinnamaneni what assumption is made here about the size of RAM heap relative to the total size of all packets sent?